pkg/render/dex.go

```diff
@@ -398,25 +408,10 @@ func (c *dexComponent) allowTigeraNetworkPolicy(installationVariant operatorv1.P
 			Protocol:    &networkpolicy.TCPProtocol,
 			Destination: networkpolicy.KubeAPIServerEntityRule,
 		},
-
-		// These rules allow egress between dex and identity providers.
-		{
-			Action:   v3.Allow,
-			Protocol: &networkpolicy.TCPProtocol,
-			Destination: v3.EntityRule{
-				Nets:  []string{"0.0.0.0/0"},
-				Ports: networkpolicy.Ports(443, 6443, 389, 636),
-			},
-		},
-		{
-			Action:   v3.Allow,
-			Protocol: &networkpolicy.TCPProtocol,
-			Destination: v3.EntityRule{
-				Nets:  []string{"::/0"},
-				Ports: networkpolicy.Ports(443, 6443, 389, 636),
-			},
-		},
 	}...)
+	for _, egressRule := range c.resolveEgressRulesByDestination() {
```

Comment:
I know that policy rules are already tested in the controller test code as well, what I am missing is a test showing a practical example of proxy settings being set in the Installation Spec and the resulting network policy, so the two can be compared side-by-side to see if it the resulting policy makes sense. I feel that would help maintaining the code if we make any changes in the future.